docs/proofs/example1.rst

```diff
@@ -0,0 +1,280 @@
+Example 1
```

Comment:
I changed the title to 'Elaborator Reflection - Identity Example' because I think that, at some stage in the future, there should be other examples added.